src/action/base/setup_default_profile.rs

```diff
@@ -94,12 +96,90 @@ impl Action for SetupDefaultProfile {
             )));
         };
 
+        {
+            let reginfo_path =
+                Path::new(crate::action::base::move_unpacked_nix::DEST).join(".reginfo");
+            let reginfo = tokio::fs::read(&reginfo_path)
+                .await
+                .map_err(|e| ActionError::Read(reginfo_path.to_path_buf(), e))?;
+            let mut load_db_command = Command::new(nix_pkg.join("bin/nix-store"));
+            load_db_command.process_group(0);
+            load_db_command.arg("--load-db");
+            load_db_command.stdin(std::process::Stdio::piped());
+            load_db_command.env(
+                "HOME",
+                dirs::home_dir().ok_or_else(|| {
+                    ActionError::Custom(Box::new(SetupDefaultProfileError::NoRootHome))
+                })?,
+            );
+            let load_db_command_str = format!("{:?}", load_db_command.as_std());
+            tracing::trace!(
+                "Executing `{load_db_command_str}` with stdin from `{}`",
+                reginfo_path.display()
+            );
+            let mut handle = load_db_command
+                .spawn()
+                .map_err(|e| ActionError::Command(e))?;
+
+            let mut stdin = handle.stdin.take().unwrap();
+            stdin
+                .write_all(&reginfo)
+                .await
+                .map_err(|e| ActionError::Write(PathBuf::from("/dev/stdin"), e))?;
+            stdin
+                .flush()
+                .await
+                .map_err(|e| ActionError::Write(PathBuf::from("/dev/stdin"), e))?;
+            drop(stdin);
+            tracing::trace!(
+                "Wrote `{}` to stdin of `nix-store --load-db`",
+                reginfo_path.display()
+            );
+
+            let output = handle
+                .wait_with_output()
+                .await
+                .map_err(ActionError::Command)?;
+            match output.status.success() {
```

Comment:
Doesn't actually matter, but could this be written more succinctly as an `if !success { return Err... }`? `match`ing on `true` or `false` makes me feel weird, but I can't really explain it :laughing:

Reply:
I am a huge fan of matching on booleans and it is not recommended against by any style guides, it's even quite common when you want to ensure both sides are always handled.. xD

Reply:
In this case your option is definitely better I think.